From 298a8dd0078012d2f37f1af8727dc1aed4fec4ab Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Mon, 19 Jun 2023 11:04:26 +0200
Subject: [PATCH 0401/1141] drm/stm: lvds: don't enable lvds pixel clock

Don't enable lvds pixel clock during probing. That's not necessary
to prepare/enable this clock & the lvds pll can be blocked.

Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Change-Id: I0e7c962104b12e887206bf3e46116479ea3c9ce3
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/313527
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/stm/lvds.c | 7 -------
 1 file changed, 7 deletions(-)

--- a/drivers/gpu/drm/stm/lvds.c
+++ b/drivers/gpu/drm/stm/lvds.c
@@ -1188,12 +1188,6 @@ static int lvds_probe(struct platform_de
 		goto err_lvds_probe;
 	}
 
-	ret = clk_prepare_enable(lvds->lvds_ck_px.clk);
-	if (ret) {
-		DRM_ERROR("%s: Failed to enable reference clk\n", __func__);
-		goto err_lvds_probe;
-	}
-
 	lvds->lvds_bridge.funcs = &lvds_bridge_funcs;
 	lvds->lvds_bridge.of_node = dev->of_node;
 
@@ -1203,7 +1197,6 @@ static int lvds_probe(struct platform_de
 
 
 err_lvds_probe:
-	clk_disable_unprepare(lvds->lvds_ck_px.clk);
 	clk_disable_unprepare(lvds->pclk);
 
 	return ret;
